ladybug/wea.py

```diff
@@ -246,48 +279,51 @@ def from_ashrae_clear_sky(cls, location, sky_clearness=1, timestep=1):
 
         # create sunpath and get altitude at every timestep of the year
         sp = Sunpath.from_location(location)
+        sp.is_leap_year = is_leap_year
         altitudes = []
         months = []
-        dates = []
-        for h in xrange(8760 * timestep):
-            if timestep == 1:
-                t_date = DateTime.from_moy((h * 60) + 30)
-            else:
-                t_date = DateTime.from_moy((h / float(timestep)) * 60)
+        dates = cls._get_datetimes(timestep, is_leap_year)
+        for t_date in dates:
             sun = sp.calculate_sun_from_date_time(t_date)
-            dates.append(sun.datetime)
             months.append(sun.datetime.month - 1)
             altitudes.append(sun.altitude)
 
         # compute hourly direct normal and diffuse horizontal radiation
         direct_norm_rad, diffuse_horiz_rad = \
-            cls._get_data_collections(location, timestep)
+            cls._get_empty_data_collections(location, timestep, is_leap_year)
 
         for i, alt in enumerate(altitudes):
-            if alt > 0.1:
```

Comment:
I think taking this out might cause an error since it was causing an exception to be thrown after it computes e raised to an extremely high exponent that cannot be expressed in 16-bit float numbers.  If it's passing the checks, though, maybe I am just being paranoid.  Let's merge and I'll do some checks of my own.

Reply:
This should handle the exception:
https://github.com/ladybug-tools/ladybug/pull/61/files#diff-f268b272e07e1ae230a4c18677efa6cbR307